ci2/ci/github.py

```diff
@@ -166,6 +170,27 @@ def config(self):
             'sha': self.sha
         }
 
+    def most_recent_build_has_out_of_date_source(self):
+        assert self.batch is None
+        return self.most_recent_batch is None or self.source_sha != self.most_recent_batch.attributes['source_sha']
+
+    def build_is_complete(self):
+        return build_state_is_complete(self.build_state)
+
+    def should_update_most_recent_batch(self):
+        return self.batch is not None and (
+            self.most_recent_batch is None or
+            self.build_is_complete() or
+            self.source_sha == self.batch.attributes['source_sha'])
```

Comment:
This last condition doesn't quite seem right.  If the most recent build is complete, and the current build is not complete, we don't want to move, do we, whether the shas match or not.